pythonforandroid/toolchain.py

```diff
@@ -1108,7 +1108,10 @@ def _build_package(self, args, package_type):
             else:
                 raise BuildInterruptingException(
                     "Unknown build mode {} for apk()".format(args.build_mode))
-            output = shprint(gradlew, gradle_task, _tail=20,
+
+            # WARNING: We should make sure to clean the build directory before building.
+            # Looks like our private.tar is causing issues to gradle.
+            output = shprint(gradlew, "clean", gradle_task, _tail=20,
```

Comment:
Nice findings and fix :muscle: 
Do you think we should reference this pull request or initial bug report with a `refs #2705` in the comment since you explained it all well?

Reply:
I was waiting to split the original issue,  into multiple chunks (as actually contains multiple issues, and some of them are still being triaged) and then totally forgot to add a reference.

So, I guess that now that we have a PR to reference, I will reference the PR. 👍